File: FriendsNumber.c
#include <stdio.h>

int extractDigits(int number, int digits[]) {
    int count = 0;
    while (number > 0) {
        digits[count++] = number % 10;
        number /= 10;
    }
    return count;
}

int isFriend(int a, int b) {
    int aDigits[10] = {0};
    int bDigits[10] = {0};
    int aCount = extractDigits(a, aDigits);
    int bCount = extractDigits(b, bDigits);

    for (int i = 0; i < aCount; i++) {
        for (int j = 0; j < bCount; j++) {
            if (aDigits[i] == bDigits[j]) {
                return 1;
            }
        }
    }
    return 0;
}

int main() {
    int end, counter = 0;
    printf("Enter the end value: ");
    scanf("%d", &end);

    for (int i = 1; i < end; i++) {
        for (int j = i + 1; j <= end; j++) {
            counter += isFriend(i, j);
        }
    }

    printf("%d friends found in the interval 1 .. %d.\n", counter, end);

    return 0;
}

File: FriendsNumber.py
def extract_digits(number):
    digits = []
    while number > 0:
        digits.append(number % 10)
        number //= 10
    return digits

def is_friend(a, b):
    a_digits = set(extract_digits(a))
    b_digits = set(extract_digits(b))
    return bool(a_digits & b_digits)

end = int(input("Enter the end value: "))  # Get user input for the 'end' value
counter = 0

for i in range(1, end):
    counter += sum(is_friend(i, j) for j in range(i + 1, end + 1))

print(f"{counter} friends found in the interval 1 .. {end}.")
